lib/active_merchant/billing/gateways/bambora_apac.rb

```diff
@@ -150,36 +178,48 @@ def new_submit_xml(action)
         xml.instruct!
         xml.soap :Envelope, 'xmlns:xsi' => 'http://www.w3.org/2001/XMLSchema-instance', 'xmlns:xsd' => 'http://www.w3.org/2001/XMLSchema', 'xmlns:soap' => 'http://schemas.xmlsoap.org/soap/envelope/' do
           xml.soap :Body do
-            xml.__send__(action, 'xmlns' => 'http://www.ippayments.com.au/interface/api/dts') do
-              xml.trnXML do
-                inner_xml = Builder::XmlMarkup.new(indent: 2)
-                yield(inner_xml)
-                xml.cdata!(inner_xml.target!)
+            xml.__send__(action, 'xmlns' => "http://www.ippayments.com.au/interface/api#{define_endpoint(action)}") do
+              if action == 'TokeniseCreditCard'
+                xml.tokeniseCreditCardXML do
+                  inner_xml = Builder::XmlMarkup.new(indent: 2)
+                  yield(inner_xml)
+                  xml.cdata!(inner_xml.target!)
+                end
+              else
+                xml.trnXML do
+                  inner_xml = Builder::XmlMarkup.new(indent: 2)
+                  yield(inner_xml)
+                  xml.cdata!(inner_xml.target!)
+                end
               end
             end
           end
         end
         xml.target!
       end
 
+      def define_endpoint(action)
+        action == 'TokeniseCreditCard' ? '/sipp' : '/dts'
+      end
```

Comment:
This won't hold up approval, just something to consider. I think I would name the function `endpoint` and avoid including the beginning forward slash in the return values - it was not immediately apparent when reviewing the URL's where it was or how it was included.

```
def endpoint(action)
  action == 'TokeniseCreditCard' ? 'sipp' : 'dts'
end

# Example usage
response = parse(ssl_post("#{commit_url}/#{endpoint(action)}.asmx", new_submit_xml(action, &block), headers))
```

Reply:
➕

Reply:
✅